src/sourcekit-lsp/getReferenceDocument.ts

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VS Code Swift open source project
+//
+// Copyright (c) 2024 the VS Code Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VS Code Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from "vscode";
+import * as langclient from "vscode-languageclient/node";
+import { GetReferenceDocumentParams, GetReferenceDocumentRequest } from "./lspExtensions";
+
+export function activateGetReferenceDocument(client: langclient.LanguageClient): vscode.Disposable {
+    const getReferenceDocument = vscode.workspace.registerTextDocumentContentProvider(
+        "swift-macro-expansion",
```

Comment:
As with the peek document can we keep this generic so it is possible to use in other situations. Maybe `reference-document`. 

@ahoppen thoughts?

Reply:
Interesting thought to just use one URL scheme. 

My thought was that we would add another URL scheme for Swift interfaces in the future and add that to the VS Code extension as well. The benefit of having different URL schemes would be that we can use the path of the URL to actually carry data. If we use one scheme, we need to use the path portion to disambiguate which kind of document we want. 

But given that we already use the path for a dual purpose (the last component is the display name to be used in VS Code), I agree that it would be good to change the URL scheme to 

```swift
sourcekit-lsp://<document-type>/<display-name>?parameters=values
```

For macro expansions, `document-type` would be `swift-macro-expansion` and the full URL would be adjusted from the proposal in https://github.com/swiftlang/sourcekit-lsp/pull/1567 to the following (`sourcekit-lsp://macro-expansion` is the only fixed portion here).

```
sourcekit-lsp://macro-expansion/LaCb-LcCd.swift?sourceFile=path_to_source_file&lowerL=&lowerC=&upperL=&upperC=&bufferName=`
```